src/modules/iotjs_module_adc.c

```diff
@@ -216,17 +198,36 @@ static void iotjs_adc_close_worker(uv_work_t* work_req) {
 
 JHANDLER_FUNCTION(AdcConstructor) {
   DJHANDLER_CHECK_THIS(object);
-  DJHANDLER_CHECK_ARGS(2, object, function);
 
   // Create ADC object
   const iotjs_jval_t* jadc = JHANDLER_GET_THIS(object);
   iotjs_adc_t* adc = iotjs_adc_create(jadc);
   IOTJS_ASSERT(adc == iotjs_adc_instance_from_jval(jadc));
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_adc_t, adc);
+
+  const iotjs_jval_t* jconfiguration = JHANDLER_GET_ARG_IF_EXIST(0, object);
+  if (jconfiguration == NULL) {
+    JHANDLER_THROW(TYPE, "Bad arguments - configuration should be Object");
+    return;
+  }
 
-  iotjs_adc_set_configuration(adc, JHANDLER_GET_ARG(0, object));
+#if defined(__linux__)
+  DJHANDLER_GET_REQUIRED_CONF_VALUE(jconfiguration, _this->device,
+                                    IOTJS_MAGIC_STRING_DEVICE, string);
+#elif defined(__NUTTX__)
+  DJHANDLER_GET_REQUIRED_CONF_VALUE(jconfiguration, _this->pin,
+                                    IOTJS_MAGIC_STRING_PIN, number);
+#endif
 
-  const iotjs_jval_t* jcallback = JHANDLER_GET_ARG(1, function);
-  ADC_ASYNC(open, adc, jcallback, kAdcOpOpen);
+  if (iotjs_jhandler_get_arg_length(jhandler) > 1) {
+    const iotjs_jval_t* jcallback = iotjs_jhandler_get_arg(jhandler, 1);
+    if (iotjs_jval_is_function(jcallback)) {
+      ADC_ASYNC(open, adc, jcallback, kAdcOpOpen);
```

Comment:
If there is no `jcallback`, ADC not open. Please change this line to execute even if it is not.